R/registry.R

```diff
@@ -34,14 +34,17 @@ is_credential_function <- function(f) {
 
 #' Add a new credential fetching function.
 #'
-#' Note that this implicitly adds `f` to the *end* of the list.
+#' Note that this implicitly adds `f` to the *front* of the list.
 #'
 #' @param f A function with the right signature. See [is_credential_function()].
 #' @family registration
 #' @export
 add_credential_function <- function(f) {
   stopifnot(is_credential_function(f))
   gargle_env$credential_functions <- c(f, gargle_env$credential_functions)
+  if (is.symbol(substitute(f))) {
```

Comment:
I can't figure out how this will capture the function names. They aren't passed in via `name = value`. It's like this:

``` r
jenny_env <- new.env(parent = emptyenv())

foo <- function(...) {
  jenny_env$functions <- c(list(...), jenny_env$functions)
}
foo(sd)
foo(mean)
names(jenny_env$functions)
#> NULL

jenny_env$functions <- list()

bar <- function(f) {
  jenny_env$functions <- c(f, jenny_env$functions)
  if (is.symbol(substitute(f))) {
    names(jenny_env$functions)[1] <- deparse(substitute(f))
  }
}
bar(sd)
bar(mean)
names(jenny_env$functions)
#> [1] "mean" "sd"
```

For example, here's how the default credential function list comes to be:

```r
set_default_credential_functions <- function() {
  add_credential_function(get_user_oauth2_credentials)
  add_credential_function(get_gce_credentials)
  add_credential_function(get_application_default_credentials)
  add_credential_function(get_travis_credentials)
  add_credential_function(get_service_account_credentials)
}
```

I'm happy to learn a less ugly way to get the names!